### PR TITLE
Add/swap limit

### PR DESCRIPTION
**Pull request by Jianzhuo**

### CODE REVIEW

File: contracts/PermissionedExchange.sol [hunk 1]
```diff
@@ -41,6 +41,7 @@ contract PermissionedExchange is Initializable, OwnableUpgradeable {
     mapping(address => bool) public exchangeController;
     /// @notice Orders: orderId => ExchangeOrder
     mapping(uint256 => ExchangeOrder) public orders;
+    uint256 public tradingLimitation;
```
Comment (nz-neooo) [inline contracts/PermissionedExchange.sol]: add comment `/// @notice Xx`for it

**Review by nz-neooo — APPROVED** [review]

File: contracts/PermissionedExchange.sol (hunk 1 above)
Comment (mzxyz) [inline contracts/PermissionedExchange.sol]: Use `tradeLimitation` to be more professional

File: contracts/PermissionedExchange.sol [hunk 2]
```diff
@@ -178,6 +203,9 @@ contract PermissionedExchange is Initializable, OwnableUpgradeable {
         if (order.tokenGet == settings.getSQToken()) {
             require(tradeQuota[order.tokenGet][msg.sender] >= _amount, 'PE005');
         }
+        if (order.tokenGive == settings.getSQToken() && tradingLimitation > 0) {
```
Comment (mzxyz) [inline contracts/PermissionedExchange.sol]: [question] unnecessary to require `tradingLimitation > 0`, if `tradingLimitation  == 0`, that means we don't allow user to purchase `kSQT`

File: publish/revertcode.json [hunk 3]
```diff
@@ -72,6 +72,7 @@
     "PE009": "order not exist",
     "PE010": "order not expired",
     "PE011": "only order sender allowed",
+    "PE012": "reach USDC trading limitation in single transaction",
```
Comment (mzxyz) [inline publish/revertcode.json]:
```suggestion
    "PE012": "reach stable coin trading limitation in a single transaction",
```

File: contracts/PermissionedExchange.sol [hunk 4]
```diff
@@ -90,6 +93,28 @@ contract PermissionedExchange is Initializable, OwnableUpgradeable {
         exchangeController[_controller] = _isController;
     }
 
+    /**
+     * @notice Set the USDC trading limitation in single transaction.
```
Comment (mzxyz) [inline contracts/PermissionedExchange.sol]:
```suggestion
     * @notice Set the stable coin trading limitation in single transaction. 
```

**Comment by mzxyz** [comment]
@Jianzhuo 
- [x] Resolve the comments
- [x] Add unit tests

**Review by mzxyz — APPROVED** [review]

**Pull request merged**